### PR TITLE
Merge OngXinBin-AnimeStorage

### LINKED ISSUE #19: Implement JSON Parser to fetch data from AniList-Data.json
Allows AniChan to be able to handle the data

### PR DESCRIPTION
**Pull request by n3wsoldier**
Branch implementation:

1) Create AnimeStorage Class
2) Dependency for JSONSimple
3) Parse AniListData JSON files
4) Store AniListData in AnimeData class

Closes #19

### CODE REVIEW

File: src/main/java/seedu/duke/AnimeStorage.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    private void parseJson(ArrayList<Anime> animeDataList,FileReader  fileData) {
+        JSONParser parser = new JSONParser();
+        JSONArray jsonList = new JSONArray();
+        try {
+            jsonList = (JSONArray) parser.parse(fileData);
+        } catch (ParseException | IOException e) {
+            e.printStackTrace();
+        }
+
+        Iterator<JSONObject> iterator = jsonList.iterator();
```
Comment (EyoWeiChin) [inline src/main/java/seedu/duke/AnimeStorage.java]: This part could be changed to: 
Iterator iterator = jsonList.iterator();

To remove the depreciated / unchecked warning

File: src/main/java/seedu/duke/AnimeStorage.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        JSONObject jsonObject = new JSONObject();
+        while (iterator.hasNext()) {
+            String animeName;
+            int animeEpisode = 0;
+            int animeRating = 0;
+            int animeDuration = 0;
+
+            jsonObject = (JSONObject) iterator.next().get("data");
```
Comment (EyoWeiChin) [inline src/main/java/seedu/duke/AnimeStorage.java]: Following the above comment, this line could be split into these 2 parts to:
            //Advance the Iterator
            jsonObject = (JSONObject) iterator.next();
            jsonObject = (JSONObject) jsonObject.get("data");

File: src/main/java/seedu/duke/AnimeStorage.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            }
+
+            //getting genre
+            ArrayList<String> animeGenre;
+            animeGenre = new ArrayList<>();
+            JSONArray jsonGenre = (JSONArray) jsonObject.get("genres");
+            Iterator<String> iteratorGenre = jsonGenre.iterator();
+            while (iteratorGenre.hasNext()) {
+                animeGenre.add(iteratorGenre.next());
+            }
```
Comment (EyoWeiChin) [inline src/main/java/seedu/duke/AnimeStorage.java]: Following the above comment to remove the warning. This chunk of code can be replaced with:
            for (Object genre : jsonGenre) {
                animeGenre.add((String) genre);
            }

**Review by EyoWeiChin — CHANGES_REQUESTED** [review]
I have included some changes which help removes the unchecked / depreciated warning. Hope it would be useful!

**Review by EyoWeiChin — APPROVED** [review]

**Review by OngDeZhi — APPROVED** [review]
LGTM :)

**Pull request merged**